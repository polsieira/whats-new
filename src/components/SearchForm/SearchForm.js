import React, { Component } from 'react';
import './SearchForm.scss';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      search: "",
    }
  }

  handleSearch = event => {
    if (event.keyCode === 13 || event.target.className === "button--search") {
      event.preventDefault();
      this.props.searchArticles(this.state.search);
      this.resetState();
    }
  }

  resetState = () => {
    this.setState({
      search: ""
    })
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    return (
      <header className="SearchForm" >
        <input
          className="input--search"
          placeholder="Search for news articles here..."
          name="search"
          value={this.state.search}
          onChange={event => this.handleChange(event)}
          onKeyDown={event => this.handleSearch(event)}>
        </input>
        <button
          className="button--search"
          onClick={event => this.handleSearch(event)}>
          Search Now
        </button>
      </header>
    )
  }
}

export default SearchForm;