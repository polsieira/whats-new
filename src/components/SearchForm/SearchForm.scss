.SearchForm {
  display: flex;
  justify-content: space-between;
  padding: 15px;
}

.input--search {
  background: none;
  border: none;
  color: #ffffff;
  font-size: 18px; 
  height: 30px;
  width: 50%;

  &:focus {
    outline: none;
  }
}

.button--search {
  background: #B56AF0;
  border: none;
  border-radius: 15px;
  color: #ffffff;
  font-size: 12px; 
  height: 30px;
  padding: 5px;
  width: 15%;

  &:hover {
    cursor: pointer;
    background: #FFAE02;
  }

  &:focus {
    outline: none;
  }
}