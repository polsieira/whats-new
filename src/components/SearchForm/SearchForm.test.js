import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  let wrapper;
  const mockSearchArticles = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<SearchForm searchArticles={mockSearchArticles} />)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to handle change in state based on an event', () => {
    //Setup
    const mockEvent = {
      target: {
        name: 'search',
        value: 'Chicken'
      }
    };

    //Execution
    wrapper.instance().handleChange(mockEvent);

    //Expectation
    expect(wrapper.state()).toEqual({ search: 'Chicken' })
  });

  it('should be able to reset its own state', () => {
    //Setup
    const newState = { search: 'Chicken' };
    const clearedState = { search: '' };

    //Execution
    wrapper.setState(newState);
    expect(wrapper.state()).toEqual(newState);

    wrapper.instance().resetState();

    //Expectation
    expect(wrapper.state()).toEqual(clearedState);
  });

  it('should handle search when search button is clicked', () => {

  })
});
