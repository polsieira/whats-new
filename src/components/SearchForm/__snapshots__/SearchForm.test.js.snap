// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with all data passed in correctly 1`] = `
<header
  className="SearchForm"
>
  <input
    className="input--search"
    name="search"
    onChange={[Function]}
    placeholder="Search for news articles here..."
    value=""
  />
  <button
    className="button--search"
    onClick={[Function]}
  >
    Search Now
  </button>
</header>
`;
